src/proxy/http/router.rs

```diff
@@ -97,6 +103,65 @@ where
     }
 }
 
+impl<Req, Rec> Clone for Layer<Req, Rec>
+where
+    Rec: Recognize<Req> + Clone + Send + Sync + 'static,
+{
```

Comment:
The previous derived `Clone` implementation for  `Layer` (and for `Stack`) required that the request type also be `Clone`, even though it's a `PhantomData`; this had to be fixed since the fallback layer requires cloning the fallback service.